modules/solid_mechanics/src/materials/HillCreepStressUpdate.C

```diff
@@ -173,14 +243,155 @@ HillCreepStressUpdateTempl<is_ad>::computeDerivative(
     qsigma_square = Ms.dot(stress_new);
   }
 
-  qsigma_square = std::sqrt(qsigma_square);
-  qsigma_square -= 1.5 * _two_shear_modulus * delta_gamma;
+  GenericReal<is_ad> qsigma = std::sqrt(qsigma_square);
+  GenericReal<is_ad> qsigma_reduced;
+
+  if (!_anisotropic_elasticity)
+    qsigma_reduced = qsigma - 1.5 * _two_shear_modulus * delta_gamma;
+  else
+  {
+    GenericRankTwoTensor<is_ad> stress;
+    stress(0, 0) = stress_new(0);
+    stress(1, 1) = stress_new(1);
+    stress(2, 2) = stress_new(2);
+    stress(0, 1) = stress(1, 0) = stress_new(3);
+    stress(1, 2) = stress(2, 1) = stress_new(4);
+    stress(0, 2) = stress(2, 0) = stress_new(5);
+
+    GenericDenseVector<is_ad> b(6);
+    b(0) = H * (stress(0, 0) - stress(1, 1)) - G * (stress(2, 2) - stress(0, 0));
+    b(1) = F * (stress(1, 1) - stress(2, 2)) - H * (stress(0, 0) - stress(1, 1));
+    b(2) = G * (stress(2, 2) - stress(0, 0)) - F * (stress(1, 1) - stress(2, 2));
+    b(3) = 2.0 * N * stress(0, 1);
+    b(4) = 2.0 * L * stress(1, 2);
+    b(5) = 2.0 * M * stress(0, 2);
+
+    GenericRankTwoTensor<is_ad> inelasticStrainIncrement;
+    inelasticStrainIncrement(0, 0) = delta_gamma * b(0) / qsigma;
+    inelasticStrainIncrement(1, 1) = delta_gamma * b(1) / qsigma;
+    inelasticStrainIncrement(2, 2) = delta_gamma * b(2) / qsigma;
+    inelasticStrainIncrement(0, 1) = inelasticStrainIncrement(1, 0) = delta_gamma * b(3) / qsigma;
+    inelasticStrainIncrement(1, 2) = inelasticStrainIncrement(2, 1) = delta_gamma * b(4) / qsigma;
+    inelasticStrainIncrement(0, 2) = inelasticStrainIncrement(2, 0) = delta_gamma * b(5) / qsigma;
+
+    stress_reduced = stress - _elasticity_tensor[_qp] * inelasticStrainIncrement;
+
+    GenericReal<is_ad> qsigma_square_reduced;
+    qsigma_square_reduced = F * (stress_reduced(1, 1) - stress_reduced(2, 2)) *
+                            (stress_reduced(1, 1) - stress_reduced(2, 2));
+    qsigma_square_reduced += G * (stress_reduced(2, 2) - stress_reduced(0, 0)) *
+                             (stress_reduced(2, 2) - stress_reduced(0, 0));
+    qsigma_square_reduced += H * (stress_reduced(0, 0) - stress_reduced(1, 1)) *
+                             (stress_reduced(0, 0) - stress_reduced(1, 1));
+    qsigma_square_reduced += 2 * L * stress_reduced(1, 2) * stress_reduced(1, 2);
+    qsigma_square_reduced += 2 * M * stress_reduced(0, 2) * stress_reduced(0, 2);
+    qsigma_square_reduced += 2 * N * stress_reduced(0, 1) * stress_reduced(0, 1);
+    qsigma_reduced = std::sqrt(qsigma_square_reduced);
+  }
```

Comment:
Instead of repeating the identical code from the residual calculation, put it in a function and call it from the residual and the derivative.

Reply:
Done. Added a dedicated function `computeQsigmaChanged` at the end that does this.